fix: remove custom Exceptions and `assert` to make scripts executable at the simplest possible setup.

lanlooker opened the pull request:
To address the question raised by our community on this error: `ModuleNotFoundError: No module named 'sdk_exceptions' `

Fixing: 
looker-open-source#756
https://community.looker.com/looker-api-77/looker-api-connection-test-26122

joeldodge79 commented on examples/python/README.md:
```diff
@@ -4,6 +4,11 @@ You can find Python language examples in this folder.
 
 The full details of all Looker API endpoints are listed in Looker Docs: [Version 3.1](https://docs.looker.com/reference/api-and-integration/api-reference/v3.1), [Version 4.0](https://docs.looker.com/reference/api-and-integration/api-reference/v4.0)
 
+## Notes
+- Many example scripts include the following statement `import sdk_exceptions`, which refers to this file [sdk_exceptions.py](sdk_exceptions.py). You could either uncomment the statement or download `sdk_exceptions.py` and put it in the same folder with your current script. 
+
+
+
```
interesting - it make sense that people are probably just copy/pasting an example or snippets from an example into their own pre-configured environment (where they maybe installed the other imports from pypi) rather than within the context of cloning this repo and executing the script with Pipfile.

@josephaxisa I'm inclined to say that no standalone example within this folder should import any other module in this folder in able to better support what is probably this common use-case. If you agree, @lanlooker instead of making this change could you delete `sdk_examples.py` and just copy/paste those error classes into each individual script? Or honestly, they could all be replaced with `raise Exception` (or maybe a more suitable stdlib one from https://docs.python.org/3/library/exceptions.html but don't spend more than a few minutes trying to find a replacement, just go with `Exception`). I don't think we need to be in the business of enforcing best patterns around specific exception handling with these examples.

The exception to the "don't import other /examples/python modules" might be if an example is a package (e.g. a folder like the current `/examples/python/{hackathon_app,lookersdk-flask} examples. Then folks should realize they need that whole folder/package structure

josephaxisa replied:
Agreed. I think making examples self contained makes perfect sense

lanlooker replied:
I see, I'll review and make changes to all of the scripts currently using the sdk_exceptions (I think most of them were written by @josephaxisa, I'll tag you so you could review them before merging)

lanlooker commented:
Hi @joeldodge79  & @josephaxisa

Currently 5 of our code snippets are using `import sdk_exceptions`: 
- Download dashboard PDF
- Download look
- Run query Task
- Run Look with Applied Filters
- Test connection

What I did: 
- Replace all custom exceptions with `raise Exception("messages")`
- Deleted all `assert` per Google's Python style guideline: `Do not use assert statements for validating argument values of a public API.` (Some of the scripts did not work when there was `assert` -- perhaps related to datatype required between API 3.0 and API 4.0. I found a new bug that the SDK 4.0 requires `id` of a dashboard to be a string, so I removed `assert` and the script works) 
- Tested these scripts to make sure they work correctly: are using API 3.0, so I initialized 4.0, and test them on an internal instance
- Added docstrings at the beginning of the scripts: I moved and rewrote docstrings of the function main() to the beginning of the scripts. Many scripts place main() at the end, so I thought it may be difficult for developers to scroll through the entire scripts to read the docstrings

To-Dos (if I have the time): 
- Write docstrings for each function, and perhaps a short summary at the top of each scripts describing the API workflow (since all scripts here use multiple API endpoints)
- Refactor the scripts: Some scripts have inconsistent datatype for parameters (i.e.: `python3 scripts.py "Title" png` -- would be nice to clarify datatypes and make them all the same)
- Review the remaining snippets 

I am using Python's style guideline to review the code: https://google.github.io/styleguide/pyguide.html

Thank you! Please let me know if you would have further comments and suggestions for me!

lanlooker commented on examples/python/download_dashboard_pdf.py:
```diff
@@ -59,10 +61,10 @@ def download_dashboard(
     """Download specified dashboard as PDF"""
-    assert dashboard.id
-    id = int(dashboard.id)
+    id = dashboard.id
```
New issue here ... API 4.0 (SDK) requires sdk.create_dashboard_render_task(dashboard_id: str) (it was int according to this script)

lanlooker commented on examples/python/download_dashboard_pdf.py:
```diff
@@ -103,4 +105,4 @@ def download_dashboard(
 
 
-main()
+main()
```
Thank you, I fixed the EOF character!

lanlooker replied:
Fixed, and merged to the current PR

lanlooker commented on examples/python/download_dashboard_pdf.py:
```diff
@@ -8,28 +21,18 @@
-import sdk_exceptions
-
-sdk = looker_sdk.init31("../../looker.ini")
+sdk = looker_sdk.init40(config_file='../../../looker.ini', section='Looker')
```
Thank you, I changed all scrips back to using this one --> `sdk = looker_sdk.init40("../looker.ini")`

(I was using one ini file for other projects so I had multiple sections in my ini file. I made a separate ini file in this example folders and gitignore it, so we should be good)

joeldodge79 commented on examples/python/download_dashboard_pdf.py:
```diff
@@ -8,28 +21,18 @@
-import sdk_exceptions
-
-sdk = looker_sdk.init31("../../looker.ini")
+sdk = looker_sdk.init40("../looker.ini")
```
oops, down one level too many
```suggestion
sdk = looker_sdk.init40("../../looker.ini")
```

joeldodge79 commented on examples/python/download_look.py:
```diff
@@ -1,28 +1,32 @@
-
-sdk = looker_sdk.init31("../../looker.ini")
-
+sdk = looker_sdk.init40("../looker.ini")
```
```suggestion
sdk = looker_sdk.init40("../../looker.ini")
```

joeldodge79 commented on examples/python/query_task.py:
```diff
@@ -6,9 +16,7 @@
-import sdk_exceptions
-
-sdk = looker_sdk.init40("../../looker.ini")
+sdk = looker_sdk.init40("../looker.ini")
```
```suggestion
sdk = looker_sdk.init40("../../looker.ini")
```

joeldodge79 commented on examples/python/run_look_with_filters.py:
```diff
@@ -1,31 +1,33 @@
-import sdk_exceptions
-
-sdk = looker_sdk.init31("../../looker.ini")
+sdk = looker_sdk.init40("../looker.ini")
```
```suggestion
sdk = looker_sdk.init40("../../looker.ini")
```

joeldodge79 commented on examples/python/test_connection.py:
```diff
@@ -1,26 +1,31 @@
-
-sdk = looker_sdk.init31("../../looker.ini")
-
+sdk = looker_sdk.init40("../looker.ini")
```
```suggestion
sdk = looker_sdk.init40("../../looker.ini")
```

joeldodge79 reviewed (changes requested):
I'll commit the suggestions and merge

joeldodge79 reviewed (approved):
awesome, thanks @lanlooker !
